crates/sncast/src/state/hashing.rs

```diff
@@ -0,0 +1,77 @@
+use base64ct::{Base64, Encoding};
+use cairo_felt::Felt252;
+use itertools::Itertools;
+use sha3::Digest;
+use sha3::Sha3_256;
+
+// if we change API this might have collisions with old API hashes
+pub fn generate_id(selector: &str, inputs: &[Felt252]) -> String {
+    let mut res = selector.as_bytes().to_owned();
+    let mut inputs_bytes: Vec<u8> = inputs.iter().flat_map(Felt252::to_bytes_be).collect_vec();
+    res.append(&mut inputs_bytes);
+    let res = res.iter().map(|item| format!("{item:x?}")).join("").clone();
+
+    let hash = Sha3_256::new().chain_update(res).finalize();
+    Base64::encode_string(&hash)
```

Comment:
is base64 necessary, after we already did sha? I'm a bit worried about special characters in ids (but maybe I shouldn't be?)

Reply:
base64 is needed to serialise hash value into string

https://github.com/RustCrypto/hashes#:~:text=If%20you%20need%20to%20serialize%20hash%20value%20into%20string%2C%20you%20can%20use%20crates%20like%20base16ct%20and%20base64ct%3A

Reply:
we can use base16 to serialise the hash value into a hex string, but it will be longer

Reply:
> I'm a bit worried about special characters in ids (but maybe I shouldn't be?)

hm, what are the possible drawbacks?

Reply:
> the string will be longer
but it's not a big difference, so we can do a safer option

Reply:
updated